spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/admin/SpannerDatabaseAdminTemplate.java

```
@@ -0,0 +1,137 @@
+import com.google.cloud.spanner.DatabaseId;
+
+import org.springframework.util.Assert;
+
+/**
+ * @author Chengyuan Zhao
+ * @author Balint Pato
+ */
+public class SpannerDatabaseAdminTemplate {
+
+	private final DatabaseAdminClient databaseAdminClient;
+
+	private final DatabaseId databaseId;
+
+	/**
+	 * Constructor that takes in the database admin client used to perform operations and the
+	 * {@link DatabaseId} object holding the project, instance, and database IDs used for all
+	 * operations. While operations can be optionally performed for a database that does not yet
+	 * exist, the project and instance IDs must already exist for Spanner.
+	 * @param databaseAdminClient the client used to create databases and execute DDL
+	 * statements.
+	 * @param databaseId the combination of Spanner Instance Id and Database Id. While
+	 * databases can be created automatically by this template, instances determine
+	 * billing and are not created automatically.
+	 */
+	public SpannerDatabaseAdminTemplate(DatabaseAdminClient databaseAdminClient,
+			DatabaseId databaseId) {
+		Assert.notNull(databaseAdminClient, "A valid database admin client is required.");
+		Assert.notNull(databaseId, "A valid database ID is required.");
+		this.databaseAdminClient = databaseAdminClient;
+		this.databaseId = databaseId;
+	}
+
+	/**
+	 * Execute the given DDL strings in order and creates the database if it does not
+	 * exist.
+	 * @param ddlStrings
+	 * @param createDatabaseIfNotExists Has no effect if the database already exists.
+	 * Otherwise, if True then the database is created and the DDL strings are executed;
+	 * the DDL is not executed if False.
+	 */
+	public void executeDDLStrings(Iterable<String> ddlStrings,
+			boolean createDatabaseIfNotExists) {
+		if (!databaseExists() && createDatabaseIfNotExists) {
+			this.databaseAdminClient
+					.createDatabase(getInstanceId(), getDatabase(), ddlStrings).waitFor();
+		}
+		else if (databaseExists()) {
```

Maybe remove the `if` part and let it fail? Otherwise, you should have an `else` that throws an exception. Without that in place and the database no existing, the method will do nothing, and the user won't even be notified with an exception.